client/jetpack-cloud/sections/partner-portal/credit-card-fields/credit-card-submit-button.tsx

```diff
@@ -1,41 +1,40 @@
 import { Button, FormStatus, useFormStatus } from '@automattic/composite-checkout';
 import { useElements, CardElement } from '@stripe/react-stripe-js';
-import { useSelect } from '@wordpress/data';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { useI18n } from '@wordpress/react-i18n';
 import debugFactory from 'debug';
 import { useMemo } from 'react';
+import { creditCardStore } from 'calypso/state/partner-portal/credit-card-form';
 import type { StripeConfiguration } from '@automattic/calypso-stripe';
 import type { ProcessPayment } from '@automattic/composite-checkout';
-import type { StoreState } from '@automattic/wpcom-checkout';
 import type { Stripe } from '@stripe/stripe-js';
-import type { I18n } from '@wordpress/i18n';
-import type { State } from 'calypso/state/partner-portal/credit-card-form/reducer';
-import type { CreditCardSelectors } from 'calypso/state/partner-portal/types';
 
 const debug = debugFactory( 'calypso:partner-portal:credit-card' );
 
 export default function CreditCardSubmitButton( {
 	disabled,
 	onClick,
-	store,
 	stripe,
 	stripeConfiguration,
 	activeButtonText,
 }: {
 	disabled?: boolean;
 	onClick?: ProcessPayment;
-	store: State;
 	stripe: Stripe | null;
 	stripeConfiguration: StripeConfiguration | null;
 	activeButtonText: string | undefined;
 } ) {
 	const { __ } = useI18n();
-	const fields: StoreState< string > = useSelect(
-		( select ) => ( select( 'credit-card' ) as CreditCardSelectors ).getFields(),
-		[]
-	);
-	const useAsPrimaryPaymentMethod: boolean = useSelect(
-		( select ) => ( select( 'credit-card' ) as CreditCardSelectors ).useAsPrimaryPaymentMethod(),
+	const { fields, useAsPrimaryPaymentMethod, errors, incompleteFieldKeys } = useSelect(
+		( select ) => {
+			const store = select( creditCardStore );
+			return {
+				fields: store.getFields(),
+				useAsPrimaryPaymentMethod: store.useAsPrimaryPaymentMethod(),
+				errors: store.getCardDataErrors(),
+				incompleteFieldKeys: store.getIncompleteFieldKeys(),
+			};
+		},
```

Comment:
This contains 2 different sets of changes:

* Combining all `creditCardStore` selectors into one `useSelect()` for brevity and predictability.
* Removing some unnecessary types and relying on type inference instead.